tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -81,3 +81,29 @@ struct s2n_stuffer_reservation* cbmc_allocate_s2n_stuffer_reservation() {
     }
     return reservation;
 }
+
+struct s2n_array* cbmc_allocate_s2n_array() {
+    struct s2n_array *array = can_fail_malloc(sizeof(*array));
+    if(array != NULL) {
+        array->mem.size = nondet_uint32_t();
+        array->mem.allocated = nondet_uint32_t();
+        ensure_s2n_blob_has_allocated_fields(&array->mem);
+        array->len = nondet_uint32_t();
+        array->element_size = nondet_uint32_t();
+    }
+    return array;
+}
+
+static int nondet_comparator(const void *a, const void *b)
+{
```

Comment:
Is there a check we can do that `a` and `b` are valid inputs?

Reply:
I added a nullness check.